perf(linter): remove unnecessary `should_run` check

camchenry opened the pull request:
`rules` was already filtered with `should_run`, so this check doesn't do anything.

camc314 commented:
### Merge activity

* **Sep 10, 4:00 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
perf(linter): remove unnecessary `should_run` check (#13639)

`rules` was already filtered with `should_run`, so this check doesn't do anything.
